flang/lib/Lower/OpenMP/ClauseProcessor.h

```diff
@@ -115,8 +115,7 @@ class ClauseProcessor {
   bool processMap(
       mlir::Location currentLocation, Fortran::lower::StatementContext &stmtCtx,
       mlir::omp::MapClauseOps &result,
-      llvm::SmallVectorImpl<const Fortran::semantics::Symbol *> *mapSyms =
-          nullptr,
+      llvm::SmallVectorImpl<const Fortran::semantics::Symbol *> *mapSyms,
```

Comment:
I had this originally, it felt a little messy to me from what I recall so I opted for this! But if you prefer the other option and it makes more sense to you, I am perfectly happy with it as well :-)

Reply:
I agree it doesn't look great, but in my opinion it's better to do this than to force the caller to pass an output list even when it won't use it because it's needed internally. Maybe we'll be able to improve things later as we keep developing this.